script/check-non-featured-docs

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# script/check-non-featured-docs: Compare the commands featured on the reference page of the site
+#                                 to the complete list of Git commands.
+#                                 Needs a checkout of the Git source at the same level as
+#                                 the checkout of the site.
+#                                 Requires Git 2.19 or later.
+#                                 Usage: GIT_VERSION=v2.23.0 SITE_VERSION=master ./script/check-non-featured-docs
+#                                 GIT_VERSION and SITE_VERSION default to HEAD
+#                                 Note: guides are also compared, but only for GIT_VERSION >= v2.18.0
+
+cd "$(dirname "$0")/.."
+
+GIT_VERSION=${GIT_VERSION:-"HEAD"}
+SITE_VERSION=${SITE_VERSION:-"HEAD"}
+
+categories=(mainporcelain \
+            ancillarymanipulators \
+            ancillaryinterrogators \
+            foreignscminterface \
+            plumbingmanipulators \
+            plumbinginterrogators \
+            synchingrepositories \
+            synchelpers \
+            purehelpers)
+
+categories_names=("Main Porcelain Commands" \
+                  "Ancillary Commands / Manipulator" \
+                  "Ancillary Commands / Interrogators" \
+                  "Interacting with Others" \
+                  "Low-level Commands / Manipulators" \
+                  "Low-level Commands / Interrogators" \
+                  "Low-level Commands / Synching Repositories" \
+                  "Low-level Commands / Synching Helpers" \
+                  "Low-level Commands / Internal Helpers")
+
+echo "Non-featured commands:"
+echo ""
+# Get featured commands
+featured_commands=$(git grep -h -o -r -E 'git[0-9a-z-]+' $SITE_VERSION app/views/shared/ref/  ':!app/views/shared/ref/_guides.html.erb' | sort)
+cd ../git/ > /dev/null
+
+for i in ${!categories[*]}
+do
+    # Get Git commands in this group
+    cat_commands[$i]=$(git grep -h -o -E "git[0-9a-z-]+[ ]+${categories[$i]}" $GIT_VERSION command-list.txt | grep -h -o -E "git[0-9a-z-]+")
+    # Count the commands
+    tot=$(echo "${cat_commands[$i]}" | wc -l | tr -d "[[:space:]]")
+    # Diff these commands against the featured commands
+    diff_commands=$(diff -u <(echo "$featured_commands") <(echo "${cat_commands[$i]}") )
+    # Count featured commands
+    featured=$(echo "$diff_commands" | grep " git" | wc -l)
+    # Count missing commands
+    missing=$(($tot-$featured))
+    # Show missing commands
+    echo "${categories_names[$i]} ($missing/$tot)"
+    echo "$diff_commands" | grep "+git" | sed 's/+/   /'
+    echo ""
+done
+cd - > /dev/null
+
+# Get featured guides
```

Comment:
The  structure of the code is quite similar for guides and commands. It would be worth trying to factorize them

Reply:
I'll work on that, thanks.